### DISCUSSION TITLE
Support for App Startup Time

### DISCUSSION THREAD

**Discussion by shivank-gupta-01** [YueChen-C/py-ios-device#161] [Q&A] [answered]
@YueChen-C 

1. We want to use `py-ios-device` for measuring `App Startup time` in iOS applications (like cold, warm, hot etc.). 
2. One way or workaround that we can figure out is to relaunch the application (or killing it) using [this](https://github.com/YueChen-C/py-ios-device#apps), and measuring the delta in time from this timestamp in the [notifications](https://github.com/YueChen-C/py-ios-device#listen-to-app-notifications).  
3. We want to know if there is some other ways for calculating App Startup Time so that:

- Eliminate the dependency on relaunching the application.
- Enable measurement during real-time user interactions, such as repeatedly closing and reopening apps, to classify starts as cold, warm, or hot.

**Comment by YueChen-C** [comment, marked as answer]
https://github.com/YueChen-C/py-ios-device?tab=readme-ov-file#get-app-launch-lifecycle
or
https://github.com/YueChen-C/py-ios-device/blob/main/demo/instrument_demo/app_lifecycle.py
Maybe it helps？
